base/sharedarray.jl

```diff
@@ -110,10 +110,10 @@ function SharedArray{T,N}(::Type{T}, dims::Dims{N}; init=false, pids=Int[])
     S
 end
 
-SharedArray(T, I::Int...; kwargs...) = SharedArray(T, I; kwargs...)
+SharedArray{T}(I::Int...; kwargs...) = SharedArray{T,length(I)}(I; kwargs...)
```

Comment:
This should become a method of `ShardArray{T}` too replacing the ones on line 33 to 39.